File: AtCoder/ABC/B/282/B.py
n, m = map(int, input().split())
bits = list()
for i in range(n):
    bit = 0
    s = input()
    for i in range(len(s)):
        if(s[i] == 'o'): bit += (1 << i)
    bits.append(bit)

cnt = 0
for i in range(n):
    for j in range(n):
        if i <= j: continue
        if (bits[i] | bits[j]) == (1 << m) - 1: cnt += 1
print(cnt)

File: AtCoder/ABC/B/282/B.rb
n, m = gets.chomp.split.map(&:to_i)
bits = Array.new
n.times do
    bit = 0
    gets.chomp.chars.each_with_index do |c, i|
        if c == 'o'
            bit += (1 << i)
        end
    end
    bits << bit
end
cnt = 0
bits.combination(2).each do |c|
    c = c.to_a
    cnt += 1 if (c[0] | c[1]) == (1 << m) - 1
end
p cnt